File: Data Structures/Trees/HuffManCoding.java
/*  
   class Node
      public  int frequency; // the frequency of this tree
       public  char data;
       public  Node left, right;
    
*/ 

void decode(String s ,Node root){
    
    Node iter = root; 
    
    for(int i = 0; i<s.length(); i++){
        if(s.charAt(i) == '0')
            iter = iter.left; 
        else
            iter = iter.right; 

        if(iter.left == null && iter.right == null){
            System.out.print(iter.data); 
            iter = root;  
        }
    }
}

File: Data Structures/Trees/HuffManCoding.cpp
/* 
The structure of the node is

typedef struct node
{
    int freq;
    char data;
    node * left;
    node * right;
    
}node;

*/


void decode_huff(node * root,string s)
{
    node * temp = root ;
    
    
    for(int i = 0; i<s.size(); i++){
        if(s[i] == '0')
            temp = temp -> left;
        else
            temp = temp -> right;
            
        if(temp -> left == NULL && temp -> right == NULL){
            cout << temp -> data; 
            temp = root; 
        }
    }
}
